spec/features/ips/filter_ip_search_by_location_spec.rb

```diff
@@ -0,0 +1,23 @@
+describe 'filtering IP addresses by dynamic search' do
+  let!(:organisation) { create(:organisation) }
+  let(:user) { create(:user, organisation: organisation) }
+  let!(:location_one) { create(:location, organisation: organisation, address: 'Apple Street') }
+  let!(:location_two) { create(:location, organisation: organisation, address: 'Banana Road') }
+  let!(:location_three) { create(:location, organisation: organisation, address: 'Citrus Lane') }
+
+  before do
+    sign_in_user user
+    visit ips_path
+  end
+
+  context 'when user types in the search field' do
+    it 'displays the location that closely matches the text field input' do
+      fill_in 'filterInput', with: 'B'
+      # page.execute_script(filterLocations())
+
+      within('div#wrapper')do
+        expect(page).to have_selector("table", count: 1)
```

Comment:
The desired behaviour is that:
Since `Banana street` has B in it and no other location does, there should only be one table, not three. 
It fails because all three are still present.

Reply:
Assertion could be that the page doesn't have the content that's been filtered?

Reply:
I could add that an additional test e.g when there are multiple matches.

I chose this assertion because I wanted to replicate the view of the filtered page as much as possible i.e a user mimicking the same actions in the current test setup should also only see one table (with Banana Street in it).